src/main/java/ita/softserve/course_evaluation_analytics/repository/impl/ComplexChartRepositoryImpl.java

```diff
@@ -0,0 +1,38 @@
+package ita.softserve.course_evaluation_analytics.repository.impl;
+
+import ita.softserve.course_evaluation_analytics.entity.GroupsCourses;
+import ita.softserve.course_evaluation_analytics.entity.UsersFeedbacks;
+import ita.softserve.course_evaluation_analytics.mapper.GroupsCoursesMapper;
+import ita.softserve.course_evaluation_analytics.mapper.UsersFeedbacksMapper;
+import ita.softserve.course_evaluation_analytics.repository.ComplexChartRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.stereotype.Repository;
+
+import javax.sql.DataSource;
+import java.util.List;
+
+@Repository
+public class ComplexChartRepositoryImpl implements ComplexChartRepository {
+
+    private NamedParameterJdbcTemplate namedParameterJdbcTemplate;
```

Comment:
make in ```final```. You should always make such variables ```final```

Reply:
But I'm changing it in the next method

Reply:
```NamedParameterJdbcTemplate``` should be defined as a Bean in your configuration, and here you need just to inject, you are not supposed to define your config here

Reply:
Got it ))